lib/posix/getopt/getopt_long.c

```diff
@@ -172,7 +172,7 @@ parse_long_options(struct getopt_state *state, char * const *nargv,
 {
 	char *current_argv, *has_equal;
 #ifdef GNU_COMPATIBLE
-	char *current_dash;
+	char *current_dash = "";
```

Comment:
Suggest to also remove 
```c
		default:
			current_dash = "";
break;
```
in the switch below, as that code is now redundant

Reply:
The root cause is explained in: https://github.com/zephyrproject-rtos/zephyr/issues/57520#issuecomment-1538130307

Reply:
I have removed `current_dash = ""` but the default switch case is required by the code standard.